plugins/rest/aws.go

```diff
@@ -278,8 +279,17 @@ func (cs *awsMetadataCredentialService) refreshFromService(ctx context.Context)
 	// to the request
 	if _, useFullPath := os.LookupEnv(ecsFullPathEnvVar); useFullPath {
 		token, tokenExists := os.LookupEnv(ecsAuthorizationTokenEnvVar)
+		// If token doesn't exist as an env var check if it exists on the file system (e.g. for pod identities)
 		if !tokenExists {
-			return errors.New("unable to get ECS metadata authorization token")
+			tokenFilePath, tokenFilePathExists := os.LookupEnv(ecsAuthorizationTokenFileEnvVar)
```

Comment:
From reading the [docs](https://docs.aws.amazon.com/sdkref/latest/guide/feature-container-credentials.html#:~:text=Note%3A%20This%20setting%20is%20an%20alternative%20to%20AWS_CONTAINER_CREDENTIALS_RELATIVE_URI%20and%20will%20only%20be%20used%20if%20AWS_CONTAINER_CREDENTIALS_RELATIVE_URI%20is%20not%20set.), seems to me `AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE ` should take precedence over `AWS_CONTAINER_AUTHORIZATION_TOKEN `.

Reply:
Good catch! I've updated the code match this behavior.